src/tree.ts

```diff
@@ -433,9 +433,7 @@ class ProjectNode extends BaseNode {
         };
 
         for (const target of pr.targets) {
-            const srcdir = target.sourceDirectory || '';
-            const relpath = path.relative(pr.sourceDirectory, srcdir);
-            addToTree(tree, relpath, target);
+            addToTree(tree, target.folder || '', target);
```

Comment:
Should it be `target.folder || relpath`?  Or does the file api always have a value for folder even if CMAKE_FOLDER is not set?

Reply:
When `CMAKE_FOLDER` is not set for a target `target.folder` doesn't have a value.

The reason I set it to empty instead of using the relative path is to not mix-n-match using the property and the relative path. So if one target doesn't have a value it gets put directly under the project (kinda how VS handles it) and if set it uses `CMAKE_FOLDER` for the path.

Reply:
What happens if the developer doesn't use CMAKE_FOLDER at all?  It seems like this would change the behavior by passing '' for all targets.

Reply:
When `CMAKE_FOLDER` isn't used at all it simply adds the target directly under the project node without any folders.

Reply:
Old view without `CMAKE_FOLDER`:
![Old Example](https://cdn.discordapp.com/attachments/488613375262195726/976898974126583848/unknown.png)
New view without `CMAKE_FOLDER`:
![New Example](https://cdn.discordapp.com/attachments/488613375262195726/976898748313661500/unknown.png)
New view with one target before `CMAKE_FOLDER` is set and one after it's set to `Source`:
![New Example2](https://cdn.discordapp.com/attachments/488613375262195726/976899344471056444/unknown.png)